xds/internal/xdsclient/bootstrap/bootstrap_test.go

```diff
@@ -1015,23 +1015,55 @@ func TestDefaultBundles(t *testing.T) {
 	if c := bootstrap.GetCredentials("insecure"); c == nil {
 		t.Errorf(`bootstrap.GetCredentials("insecure") credential is nil, want non-nil`)
 	}
+
+	if c := bootstrap.GetCredentials("tls"); c == nil {
+		t.Errorf(`bootstrap.GetCredentials("tls") credential is nil, want non-nil`)
+	}
 }
 
 func TestCredsBuilders(t *testing.T) {
 	b := &googleDefaultCredsBuilder{}
-	if _, err := b.Build(nil); err != nil {
+	if _, stop, err := b.Build(nil); err != nil {
 		t.Errorf("googleDefaultCredsBuilder.Build failed: %v", err)
+	} else {
+		stop()
 	}
 	if got, want := b.Name(), "google_default"; got != want {
 		t.Errorf("googleDefaultCredsBuilder.Name = %v, want %v", got, want)
 	}
 
 	i := &insecureCredsBuilder{}
-	if _, err := i.Build(nil); err != nil {
+	if _, stop, err := i.Build(nil); err != nil {
 		t.Errorf("insecureCredsBuilder.Build failed: %v", err)
+	} else {
+		stop()
 	}
 
 	if got, want := i.Name(), "insecure"; got != want {
 		t.Errorf("insecureCredsBuilder.Name = %v, want %v", got, want)
 	}
+
+	tcb := &tlsCredsBuilder{}
+	if _, stop, err := tcb.Build(nil); err != nil {
+		t.Errorf("tlsCredsBuilder.Build failed: %v", err)
+	} else {
+		stop()
+	}
+	if got, want := tcb.Name(), "tls"; got != want {
+		t.Errorf("tlsCredsBuilder.Name = %v, want %v", got, want)
+	}
```

Comment:
Can we make this a table driven test instead? That way, with `t.Run`, we would be able to use `t.Fatalf` and avoid calling `stop` in the else block.

Reply:
Yes, done.